client/src/features/visualizer-threejs/ConfigControls.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from "react";
+import {
+    MIN_SINUSOID_PERIOD,
+    MAX_SINUSOID_PERIOD,
+    MIN_SINUSOID_AMPLITUDE,
+    MAX_SINUSOID_AMPLITUDE,
+    MIN_TILT_FACTOR_DEGREES,
+    MAX_TILT_FACTOR_DEGREES,
+    TILT_DURATION_SECONDS,
+    features,
+} from "./constants";
+import "./Controls.scss";
+import { useTangleStore } from "~features/visualizer-threejs/store";
+
+interface IControlsVisualiser {
+    MIN_SINUSOID_PERIOD: number;
+    MAX_SINUSOID_PERIOD: number;
+    MIN_SINUSOID_AMPLITUDE: number;
+    MAX_SINUSOID_AMPLITUDE: number;
+    MIN_TILT_FACTOR_DEGREES: number;
+    MAX_TILT_FACTOR_DEGREES: number;
+    TILT_DURATION_SECONDS: number;
+}
+
+const defaultControlsVisualiser: IControlsVisualiser = {
+    MIN_SINUSOID_PERIOD: MIN_SINUSOID_PERIOD,
+    MAX_SINUSOID_PERIOD: MAX_SINUSOID_PERIOD,
+    MIN_SINUSOID_AMPLITUDE: MIN_SINUSOID_AMPLITUDE,
+    MAX_SINUSOID_AMPLITUDE: MAX_SINUSOID_AMPLITUDE,
+    MIN_TILT_FACTOR_DEGREES: MIN_TILT_FACTOR_DEGREES,
+    MAX_TILT_FACTOR_DEGREES: MAX_TILT_FACTOR_DEGREES,
+    TILT_DURATION_SECONDS: TILT_DURATION_SECONDS,
+};
+
+type TKey = keyof IControlsVisualiser;
+
+/**
+ * Retrieves a value from localStorage and parses it as JSON.
+ */
+const LOCAL_STORAGE_KEY = "controlsVisualiser";
+
+export const getFromLocalStorage = (): IControlsVisualiser => {
+    if (features.controlsVisualiserEnabled) {
+        const item = localStorage.getItem(LOCAL_STORAGE_KEY);
+        return item ? JSON.parse(item) : defaultControlsVisualiser;
+    } else {
+        localStorage.removeItem(LOCAL_STORAGE_KEY);
+        return defaultControlsVisualiser;
+    }
+};
+
+/**
+ * Saves a value to localStorage as a JSON string.
+ */
+function setToLocalStorage(value: IControlsVisualiser) {
+    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(value));
+}
+
+/**
+ */
+function isExistsInLocalStorage(): boolean {
+    return !!localStorage.getItem(LOCAL_STORAGE_KEY);
+}
+
+export const ConfigControls = () => {
+    const forcedZoom = useTangleStore((state) => state.forcedZoom);
+    const setForcedZoom = useTangleStore((state) => state.setForcedZoom);
+    const [localZoom, setLocalZoom] = useState<number | undefined>(forcedZoom);
+
+    const [state, setState] = useState<IControlsVisualiser>(() => {
+        // Use getFromLocalStorage to retrieve the state
+        return getFromLocalStorage() || defaultControlsVisualiser;
+    });
+    const [isShowReset, setIsShowReset] = useState(() => {
+        return isExistsInLocalStorage();
+    });
+
+    const [errors, setErrors] = useState<{
+        [k: string]: string;
+    }>({});
+
+    const inputs: {
+        key: TKey;
+        label: string;
+        min: number;
+        max: number;
+    }[] = [
+        {
+            key: "MIN_SINUSOID_PERIOD",
```

Comment:
Maybe it's better to not cap the values, or set a low min and a big max value, so they can play more with the scene to get how they want it